projects/engine-vscode/src/webview.ts

```diff
@@ -0,0 +1,140 @@
+import { PluginConnector, Profile, Message, PluginConnectorOptions, ExternalProfile } from "@remixproject/engine"
+import { ExtensionContext, ViewColumn, Webview, WebviewPanel, window, Uri } from 'vscode'
+import { join, parse as parsePath } from 'path'
+import { promises as fs } from 'fs'
+import { get } from 'https'
+import { parse as parseUrl } from 'url'
+
+interface WebviewOptions extends PluginConnectorOptions {
+  /** Extension Path */
+  context: ExtensionContext
+  column?: ViewColumn
+}
+
+export class WebviewPlugin extends PluginConnector {
+  panel?: WebviewPanel
+  options: WebviewOptions
+
+  constructor(profile: Profile & ExternalProfile, options: WebviewOptions) {
+    super(profile)
+    this.options = options
+  }
+
+  setOptions(options: WebviewOptions) {
+    super.setOptions(options)
+  }
+
+  protected send(message: Partial<Message>): void {
+    if (this.panel) {
+      this.panel.webview.postMessage(message)
+    }
+  }
+
+  protected connect(url: string): void {
+    if (this.options.context) {
+      const { extensionPath } = this.options.context
+      this.panel = createWebview(this.profile, url, extensionPath, this.options.column)
+      this.panel.webview.onDidReceiveMessage(msg => this.getMessage(msg))
+      this.options.context.subscriptions.push(this.panel)
```

Comment:
can you give more context on what is `subscriptions` ?

Reply:
It's a vscode stuff to keep track of the panel.